FSQLocationBrokerTests/FSQCurrentLocationTests.m

```diff
@@ -0,0 +1,175 @@
+
+@property (nonatomic, strong, readwrite) CLLocationManager *locationManger;
+
+@property (nonatomic, strong, readwrite) CLLocation *location1;
+@property (nonatomic, strong, readwrite) CLLocation *location2;
+@property (nonatomic, strong, readwrite) CLLocation *location3;
+
+@end
+
+@implementation FSQCurrentLocationTests
+
+- (void)setUp {
+    [super setUp];
+    
+    self.locationBroker = [[FSQLocationBroker alloc] init];
+    
+    self.locationManger = [[CLLocationManager alloc] init];
+    
+    NSDate *futureDate = [NSDate dateWithTimeIntervalSinceNow:3600];
+    NSDate *beforeFutureDate = [futureDate dateByAddingTimeInterval:-100];
+    NSDate *afterFutureDate = [futureDate dateByAddingTimeInterval:603];
+    
+    CLLocationCoordinate2D pennStationCoordinates = CLLocationCoordinate2DMake(40.7484, -73.9857);
+    CLLocationCoordinate2D empireStateBuildingCoordinates = CLLocationCoordinate2DMake(40.7484, -73.9857);
+    CLLocationCoordinate2D foursquareNYCCoordinates = CLLocationCoordinate2DMake(40.7592, -73.9846);
+
+    self.location1 = [[CLLocation alloc] initWithCoordinate:pennStationCoordinates altitude:0 horizontalAccuracy:125 verticalAccuracy:0 timestamp:beforeFutureDate];
+    self.location2 = [[CLLocation alloc] initWithCoordinate:empireStateBuildingCoordinates altitude:0 horizontalAccuracy:65 verticalAccuracy:0 timestamp:futureDate];
+    self.location3 = [[CLLocation alloc] initWithCoordinate:foursquareNYCCoordinates altitude:0 horizontalAccuracy:5 verticalAccuracy:0 timestamp:afterFutureDate];
+}
+
+- (void)tearDown {
+    self.locationBroker = nil;
+
+    [super tearDown];
+}
+
+- (void)testCurrentLocationWithInOrderDates {
+    NSComparisonResult result;
+    
+    NSArray<CLLocation *> *locations = @[self.location1, self.location2, self.location3];
+    
+    [self.locationBroker locationManager:self.locationManger didUpdateLocations:locations];
+    
+    result = [self.locationBroker.currentLocation.timestamp compare:self.location3.timestamp];
+    XCTAssert(result == NSOrderedSame, @"expected most recent location");
+}
+
+- (void)testCurrentLocationWithOutOfOrderDates {
+    NSComparisonResult result;
+
+    NSArray<CLLocation *> *locations = @[self.location3, self.location1, self.location2];
+    
+    [self.locationBroker locationManager:self.locationManger didUpdateLocations:locations];
+    
+    result = [self.locationBroker.currentLocation.timestamp compare:self.location3.timestamp];
+    XCTAssert(result == NSOrderedSame, @"expected most recent location");
+}
+
+- (void)testCurrentLocationWithReverseDates {
+    NSComparisonResult result;
+
+    NSArray<CLLocation *> *locations = @[self.location3, self.location2, self.location1];
+    
+    [self.locationBroker locationManager:self.locationManger didUpdateLocations:locations];
+    
+    result = [self.locationBroker.currentLocation.timestamp compare:self.location3.timestamp];
+    XCTAssert(result == NSOrderedSame, @"expected most recent location");
```

Comment:
XCTAssertEqual does not work with NSDate or CLLocation objects. The other problem here is that both `currentLocation` and `timestamp` on CLLocation are both copy properties meaning we can not use pointer equality either. 

This was the quickest way to test we are getting expected results, the best alternative in my mind is adding a category that implements our own `isEqual:` method on CLLocation where we compare individual property values. Thoughts?